applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.h

```diff
@@ -0,0 +1,352 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Armin Geiser, https://github.com/armingeiser
+//
+
+#if !defined(ADJOINT_FINITE_DIFFERENCE_BASE_ELEMENT_H_INCLUDED )
+#define  ADJOINT_FINITE_DIFFERENCE_BASE_ELEMENT_H_INCLUDED
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/element.h"
+#include "../../custom_elements/shell_thin_element_3D3N.hpp"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+///@}
+
+///@name Type Definitions
+///@{
+///@}
+
+///@name  Enum's
+///@{
+///@}
+
+///@name  Functions
+///@{
+///@}
+
+///@name Kratos Classes
+///@{
+
+/** \brief AdjointFiniteDifferencingBaseElement
+ *
+ * This element is inherited from ShellThinElement3D3N. It is the corresponding
```

Comment:
I know, sorry about this naming inconvenience. As this is just a test, i did not change the naming anymore.
The AdjointFiniteDifferencingBaseElement should work for all  "standard" elements and would only have a element base pointer. This would be called AdjointFiniteDifferencingShell3D3NElement and derive from AdjointFiniteDifferencingBaseElement.